packages/e2e/src/FaucetProvider/types.ts

```diff
@@ -0,0 +1,89 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { HealthCheckResponse, Provider } from '../../../core';
+
+/**
+ * Faucet request transaction status.
+ */
+export enum FaucetRequestTransactionStatus {
+  Pending,
+  Submitted,
+  InLedger,
+  Expired
+}
+
+/**
+ * Result for a faucet request.
+ */
+export class FaucetRequestResult {
+  /**
+   * The transaction id of the transaction generated by this request.
+   */
+  txId: string;
+
+  /**
+   * Current status of the transaction generated by this request in the local chain .
+   */
+  status: FaucetRequestTransactionStatus;
+
+  /**
+   * Absolute time at which the transaction generated by this reuqest was inserted in a block.
+   */
+  time: string;
+
+  /**
+   * How many blocks has passed since our transaction was added to the Cardano blockchain. The more
+   * confirmations that have occured, the more secure the transaction is.
+   */
+  confirmations: number;
+}
+
+/**
+ * The faucet provider enable clients to request arbitrary amounts of tAda from current local network
+ * to fund any address. This is useful for the setup of preconditions for end-to-end tests.
+ */
+export interface FaucetProvider extends Provider {
```

Comment:
I think it'd better to reserve the term 'Provider' for the ones defined in `core` package. Maybe `RunnableModule` from `cardano-services` fits here instead?
```suggestion
export interface FaucetService extends RunnableModule {
```